monitoring/rid_qualifier/flight_data_generator.py

```diff
@@ -0,0 +1,446 @@
+        country_code: An ISO 3166-1 alpha-3 code for a country
+
+        Outputs: 
+        GeoJSON files for bboxes created in the `test_definitions/{country_code}` folder 
+        
+        
+    """
+
+    def __init__(self, grid_tracks: List[GridCellFlight], bboxes: List[QueryBoundingBox], country_code='CHE') -> None:
+        ''' This class uses the same output directory as the AdjacentCircularFlightsSimulator class and requires the path points (Tracks) and the bounding boxes from that class.
+
+        '''
+        
+        self.grid_cells_flight_tracks = grid_tracks
+        self.bboxes = bboxes
+        self.country_code = country_code        
+        self.output_directory = Path('test_definitions', self.country_code)
+        self.output_directory.mkdir(parents=True, exist_ok=True) # Create test_definition directory if it does not exist
+        self.output_subdirectories = (Path(self.output_directory, 'tracks'), Path(self.output_directory, 'query_bboxes'),)
+                                      
+        for output_subdirectory in self.output_subdirectories:
+            output_subdirectory.mkdir(parents=True, exist_ok=True) 
+
+    def write_bboxes(self):
+
+        ''' This module writes the bboxes as a GeoJSON FeatureCollection '''
+        for buffered_bbox_details in self.bboxes:
+            
+            features = json.dumps({'type': 'Feature', 'properties': {"timestamp_before":buffered_bbox_details.timestamp_before.isoformat(), "timestamp_after":buffered_bbox_details.timestamp_after.isoformat()}, 'geometry': shapely.geometry.mapping(buffered_bbox_details.shape)})
+            bbox_file_name = 'box_%s.geojson'% buffered_bbox_details.name
+            output_subdirectory = Path(self.output_directory , 'query_bboxes')
+            bbox_output_path = output_subdirectory / bbox_file_name
+
+            with open(bbox_output_path,'w') as f:
+                f.write(features)
+                
+    def write_tracks(self):
+
+        ''' This module writes tracks as a GeoJSON FeatureCollection (of Point Feature) for use in other software '''       
+        
+        flight_point_current_index = {}
+        num_flights = len(self.grid_cells_flight_tracks)
+        for i in range(num_flights):
+            flight_point_current_index[i] = 0
+            
+        for track_id, grid_cell_flight_track in enumerate(self.grid_cells_flight_tracks):
+            feature_collection = {"type":"FeatureCollection", "features": []}
+            for cur_track_point in grid_cell_flight_track.track:                
+                p = Point((cur_track_point.lat, cur_track_point.lng, cur_track_point.alt))
+                point_feature = {'type': 'Feature', 'properties': {}, 'geometry': shapely.geometry.mapping(p)}                
+                feature_collection['features'].append(point_feature)
+
+            path_file_name = 'track_%s.geojson'% str(track_id+1)
```

Comment:
Yes correct, the goal is to able to share individual flight paths, since potentially UTMSPs might need to see their flight (not others) in a way that can be shared, that's why I created individual files, I don't mind if we keep it as a single file. Do you feel strongly about this? Can we make this a priority 2 issue to tackle after the rest of the system is in place?

Reply:
I don't feel strongly -- as long as we're achieving the desired result or making progress toward the desired result, I'm fine :)

Reply:
:)